app/main/services.py

```diff
@@ -16,14 +19,79 @@ def dashboard():
     return render_template(
         "services/dashboard.html",
         services=suppliers_services["services"],
+        updated_service_id=request.args.get('updated_service_id'),
+        updated_service_name=request.args.get('updated_service_name'),
+        updated_service_status=request.args.get('updated_service_status'),
         **template_data), 200
 
 
-@main.route('/services/<string:service_id>')
+@main.route('/services/<string:service_id>', methods=['GET'])
 @login_required
 def services(service_id):
     template_data = main.config['BASE_TEMPLATE_DATA']
+    service_data = data_api_client.get_service(service_id).get('services')
+    if service_data['frameworkName'] == 'G-Cloud 5':
+        service_id = [service_id]
+    else:
+        service_id = re.findall("....", str(service_id))
+    return render_template(
+        "services/service.html",
+        service_id=service_id,
+        service_data=service_data,
+        **template_data), 200
+
+
+# Might have to change the route if we're generalizing this to update
+@main.route('/services/<string:service_id>', methods=['POST'])
+@login_required
+def update_service_status(service_id):
+
+    # Value should be either public or private
+    status = request.form['service_status']
+
+    translate_frontend_to_api = {
+        'public': 'published',
+        'private': 'enabled'
+    }
+
+    if status in translate_frontend_to_api.keys():
+        status = translate_frontend_to_api[status]
+    else:
+        return _update_service_status_error(
+            service_id,
+            "Sorry, but '{}' is not a valid status.".format(status)
+        )
+
+    try:
+        updated_service = data_api_client.update_service_status(
```

Comment:
It could - the dictionary here allows a service to be set to 'public' or 'private'.  So if Nicole has set a service to 'removed' because it is in breach of something, the supplier could hit this endpoint to change the status back to 'public' or 'private'.  The dictionary will prevent services from being 'removed', but doesn't stop them being turned back on again.
There needs to be a check on the current status of the service, and if it is removed/disabled then the update should not be done.